src/sqlfluff/core/parser/segments/indent.py

```diff
@@ -1,30 +1,24 @@
 """Indent and Dedent classes."""
 
 from ..match_wrapper import match_wrapper
+from ..markers import FilePositionMarker
 
 from .raw import RawSegment
 
 
-class Indent(RawSegment):
-    """A segment which is empty but indicates where an indent should be.
-
-    This segment is always empty, i.e. it's raw format is '', but it indicates
-    the position of a theoretical indent which will be used in linting
-    and reconstruction. Even if there is an *actual indent* that occurs
-    in the same place this intentionally *won't* capture it, they will just
-    be compared later.
-    """
+class MetaSegment(RawSegment):
+    """A segment which is empty but indicates where something should be."""

```

Comment:
Add a note about how it's used as a placeholder for pre-templated code?

Reply:
Excellent idea, I think this is also related to your comment before of:

> There are some pre-templating rules like "have a space between {{ and }} and the expression" which I'm not sure could be supported with the slices?

I'll extend the same so that it contains a reference to the source string in case anyone wants to lint it.